### PR TITLE
add message to QGIS status bar on selection of vertices

### PR DESCRIPTION
**Pull request by JanCaha**
## Description

Display message in status bar when vertices of Mesh are selected.

### CODE REVIEW

**Review by uclaros — COMMENTED** [review]
Nice!
Also keep in mind that once #58795 gets merged, `QgsDistanceArea::measureLine()` might throw a `QgsCsException` which will need to be handled.

File: src/app/mesh/qgsmaptooleditmeshframe.cpp [hunk 1]
```diff
@@ -2841,3 +2846,58 @@ void QgsMapToolEditMeshFrame::showSelectByExpressionDialog()
...
+      const QgsMeshVertex vertex2 = mesh.vertex( vertexId2 );
+
+      QString formattedDistance;
+      double distance;
+      // if crs is valid calculate using QgsDistanceArea otherwise calculate just as distance
+      if ( mCurrentLayer->crs().isValid() )
+      {
+        QgsDistanceArea distArea = QgsDistanceArea();
+        distArea.setSourceCrs( mCurrentLayer->crs(), QgsProject::instance()->transformContext() );
+        distArea.setEllipsoid( QgsProject::instance()->ellipsoid() );
```
[the remark this answers is not in the document]
Comment (nyalldawson) [reply src/app/mesh/qgsmaptooleditmeshframe.cpp:2878]: No, I think this is correct as is. If users need purely cartesian distances then they should set the project ellipsoid to "none" accordingly.

File: src/app/mesh/qgsmaptooleditmeshframe.cpp [hunk 2]
```diff
@@ -2870,17 +2870,25 @@ void QgsMapToolEditMeshFrame::updateStatusBarMessage() const
...
+          distance = distArea.measureLine( QgsPointXY( vertex1 ), QgsPointXY( vertex2 ) );
+          distance = distArea.convertLengthMeasurement( distance, QgsProject::instance()->distanceUnits() );
+          formattedDistance = distArea.formatDistance( distance, 6, QgsProject::instance()->distanceUnits() );
+          distanceIsValid = true;
+        }
+        catch ( QgsCsException & ) {}
       }
-      else
+
+      if ( ! distanceIsValid )
```
Comment (uclaros) [inline src/app/mesh/qgsmaptooleditmeshframe.cpp:2891]: You can check `if ( formattedDistance.isEmpty() )` avoiding the extra bool

**Review by nyalldawson — APPROVED** [review]

**Comment by nyalldawson** [comment]
Look ok for you @uclaros ?

**Review by uclaros — APPROVED** [review]

**Pull request merged**